src/words/fast.jl

```diff
@@ -0,0 +1,110 @@
+# TODO: the input + idx combination should probably be replaced by a stream
+
+mutable struct TokenBuffer
+  input::Vector{Char}
+  buffer::Vector{Char}
+  tokens::Vector{String}
+  idx::Int
+end
+
+TokenBuffer(input) = TokenBuffer(input, [], [], 1)
+
+TokenBuffer(input::AbstractString) = TokenBuffer(collect(input))
+
+Base.getindex(ts::TokenBuffer, i = ts.idx) = ts.input[i]
+isdone(ts::TokenBuffer) = ts.idx > length(ts.input)
+
+function flush!(ts::TokenBuffer, s...)
```

Comment:
I have a preference against single letter variable names.
because you can't **ctrl+f** them

Also  I am not sure what `s` is here. I guess tokens.

Reply:
At least in atom you can do a whole-word search